### PR TITLE
Shant: tasks 151 160

### PR DESCRIPTION
**Pull request by hsp-shant**
Please check the files for further investigations.

### CODE REVIEW

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_160.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * 
+ * @author Shant
+ *
+ */
+public class Task_160 extends TaskBase {
+
+	private final int MINIMUM_HUNDREDS_NUMBER = 100;
+	private final int MAXIMUM_HUNDREDS_NUMBER = 999;
+	private final int DIVISION_ON_NUMBER = 16;
+
```
Comment (hsp-vahe) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_160.java]: bad variable name

Comment (hsp-shant) [reply src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_160.java]: Please could you be more specific about which method?

Comment (hsp-vahe) [reply src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_160.java]: not method. variable
DIVISION_ON_NUMBER

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 2]
```diff
@@ -23,103 +24,165 @@ protected int getUserInput(String message) {
...
 	 */
-	protected boolean isMathProgression(int number1, int number2, int number3, int number4) {
-		return isMathProgression(number1, number2, number3) && isMathProgression(number2, number3, number4);
+	protected boolean isMultiplier(int number1, int number2) {
+		return number1 % number2 == 0;
+
+	}
+
+	/**
+	 * Check if the number is Multiple on another number
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: Wrong comment

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 3]
```diff
@@ -23,103 +24,165 @@ protected int getUserInput(String message) {
...
+	 */
+	protected boolean isNotMultiplier(int number1, int number2) {
+		return !isMultiplier(number1, number2);
+
+	}
+
+	/**
+	 * Check if is division by balance
+	 */
+	protected boolean isDivisionByBalance(int number1, int number2, int number3) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: Bad param names. It is not clear which param is a number, which one is a divisor, and which one is a balance...

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 4]
```diff
@@ -23,103 +24,165 @@ protected int getUserInput(String message) {
...
 	}
-		
+
 	/**
-	 * Checking mathematically Progression
+	 * Checking if number is correct number
 	 */
-	protected boolean isMathProgression(int number1, int number2, int number3) {
-		return number3 - number2 == number2 - number1 ;
+	protected boolean isDiviser(int number1, int number2) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: DivisOr

File: src/main/java/am/hsp/apps/seua/base/TaskDigits.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+	 * Method getting the Thousents of the number
+	 */
+	protected int getThousandsDigit(int number) {
+		return getDigitByIndex(number, THOUSANDS_NUMBER);
+	}
+
+	/**
+	 * Count the number of the digits in the number
+	 */
+	protected int getCountTheDigits(int number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskDigits.java]: getCount**Of**Digits

File: src/main/java/am/hsp/apps/seua/base/TaskDigits.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+		return count;
+	}
+
+	/**
+	 * Get Sum of Digits
+	 */
+	protected int getSumOfDigits(int number) {
+		int sum = getUnitsDigit(number);
+		while (number > 0) {
+			number = number / 10;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskDigits.java]: DRY violation

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package main.java.am.hsp.apps.seua.Tasks_151_160;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java]: As I told you several times - package names MUST NOT START with UPPERCASE LETTER.

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_153.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+		int number = getUserInput("Please enter your number");
+		System.out.println(getResult(number));
+
+	}
+
+	/**
+	 * Print the sum of all correct numbers, which dividing with the input number
+	 * will results as remnant the number 2.
+	 */
+	private int getResult(int number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_153.java]: same

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_154.java [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+import main.java.am.hsp.apps.seua.base.TaskBase;
+
+/**
+ * Print the Multiplication of all the correct number which dividing with the
+ * input number will results number 3 as remnant.
+ * 
+ * @author Shant
+ *
+ */
+public class Task_154 extends TaskBase {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_154.java]: same issues here

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_155.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+		
+		System.out.println(getResult());
+
+	}
+
+	/**
+	 * Print the correct numbers in the range of Tenss which are Multiple of number
+	 * 3
+	 */
+	private int getResult() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_155.java]: same

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_155.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+ * Print the sum of correct numbers in the range of Tens which are Multiple of number 3
+ * 
+ * @author Shant
+ *
+ */
+public class Task_155 extends TaskBase {
+	
+	private final int MINIMUM_TENS_NUMBER = 10;
+	private final int MAXIMUM_TENS_NUMBER = 99;
+	private final int SUCCESS_NUMBER = 3;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_155.java]: DIVISOR

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+ * Multiple of the number 3 and 5
+ * 
+ * @author Shant
+ *
+ */
+public class Task_156 extends TaskBase {
+
+	private final int MINIMUM_TENS_NUMBER = 10;
+	private final int MAXIMUM_TENS_NUMBER = 99;
+	private final int SUCCESS_FIRST_NUMBER = 3;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java]: same

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+	@Override
+	public void solve() {
+		System.out.println(getResult());
+	}
+
+	/**
+	 * Print the Multiplication of correct numbers in the range of Tens which are
+	 * Multiple of the number 3 and 5
+	 */
+	private long getResult() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java]: same

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_157.java [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+import main.java.am.hsp.apps.seua.base.TaskBase;
+
+/**
+ * Print the correct numbers in the range of Hundreds which are not Multiple of
+ * number 5
+ * 
+ * @author Shant
+ *
+ */
+public class Task_157 extends TaskBase {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_157.java]: same

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_159.java [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Print the division of the Hundreds digits for the correct number which by
+ * dividing on number 3 will renman number 1 and dividing on number 4 will
+ * renman number 2
+ * 
+ * @author Shant
+ *
+ */
+public class Task_159 extends TaskBase {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_159.java]: same

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java (hunk 7 above)
Comment (hsp-harut) [reply src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java]: You didn't fix my comment related **WRONG PACKAGE NAME**

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+
+	/**
+	 * Print the sum of all the correct numbers on which there is no remnant given
+	 * to 'N' number
+	 */
+	private int getSumOfDivisor(int number) {
+
+		int sum = 1 + number;
+
+		for (int i = MINUMUM_DIVISOR_NUMBER; i < number / 2; i++) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java]: Wrong. Your program is not working correctly, you didn't even check it?

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_152.java [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+	}
+
+	/**
+	 * Print the Multiplication of all the correct numbers on which there is no
+	 * remnant given to 'N' number
+	 */
+	private int getMulitplyOfDivisor(int number) {
+
+		int mulitplication = 1 * number;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_152.java]: 1 * number = number, so why don't have just int mulitplication = number;

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+ * Multiple of the number 3 and 5
+ * 
+ * @author Shant
+ *
+ */
+public class Task_156 extends TaskBase {
+
+	private final int MINIMUM_TENS_NUMBER = 10;
+	private final int MAXIMUM_TENS_NUMBER = 99;
+	private final int BALANCE_FIRST_NUMBER = 3;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java]: keep it simple, just BALANCE_1, BALANCE_2

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Print the Multiplication of correct numbers in the range of Tens which are
+ * Multiple of the number 3 and 5
+ * 
+ * @author Shant
+ *
+ */
+public class Task_156 extends TaskBase {
+
+	private final int MINIMUM_TENS_NUMBER = 10;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java]: There is no need to append "NUMBER" after each numeric constant. Please update **everywhere**

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+	@Override
+	public void solve() {
+		System.out.println(getMulitplyOfDivisor());
+	}
+
+	/**
+	 * Print the Multiplication of correct numbers in the range of Tens which are
+	 * Multiple of the number 3 and 5
+	 */
+	private long getMulitplyOfDivisor() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_156.java]: You ar not calculating a product of Divisors, you are calculating a product of Miltipliers

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_157.java [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+
+		System.out.println(getResult());
+
+	}
+
+	/**
+	 * Print the sum of correct numbers in the range of Hundreds which are not Multiple of
+	 * number 5
+	 */
+	private int getResult() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_157.java]: make descriptive method name

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_158.java [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+		System.out.println(getMulitplyOfDivisor());
+
+	}
+
+	/**
+	 * Print the Muliplication of correct numbers in the range of Hundreds which are
+	 * not Multiple of the number 2 and 3
+	 */
+	private long getMulitplyOfDivisor() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_158.java]: wrong method name

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_159.java [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+
+	private final int MINIMUM_HUNDREDS_NUMBER = 100;
+	private final int MAXIMUM_HUNDREDS_NUMBER = 999;
+	private final int SUCCESS_FIRST_NUMBER = 1;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_159.java]: You are not fixing my comments related wrong constant names. 
It says nothing "success first number" ... second number... third number - did it say something???

Actually these numbers are divisors and corresponding balances (remnants)

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 24]
```diff
@@ -23,103 +24,165 @@ protected int getUserInput(String message) {
...
 	}
-		
+
 	/**
-	 * Checking mathematically Progression
+	 * Checking if number is correct number
 	 */
-	protected boolean isMathProgression(int number1, int number2, int number3) {
-		return number3 - number2 == number2 - number1 ;
+	protected boolean isDivisor(int number1, int divisor) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: why number1?

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 25]
```diff
@@ -23,103 +24,165 @@ protected int getUserInput(String message) {
...
-	protected boolean isGeoProgresion(int number1, int number2, int number3, int number4) {
-		return isGeoProgresion(number1, number2, number3)&& isGeoProgresion(number2, number3, number4);
+	protected double divide(int number1, int number2) {
+		return (double) number1 / number2;
 	}
-		
+
 	/**
-	 * Checking mathematically Progression
+	 * Checking if number is correct number
```
[the remark this answers is not in the document]
Comment (hsp-harut) [reply src/main/java/am/hsp/apps/seua/base/TaskBase.java]: Again, wrong comment. Your method checks whether is the "divisor" parameter is actually a divisor for the given number...

File: src/main/java/am/hsp/apps/seua/base/TaskBase.java [hunk 26]
```diff
@@ -23,103 +24,165 @@ protected int getUserInput(String message) {
...
 	}
-	
+
 	/**
-	 * Method to Check if the numbers are Mathematically progression
+	 * Check if the number is Multiple on another number
 	 */
-	protected boolean isMathProgression(int number1, int number2, int number3, int number4) {
-		return isMathProgression(number1, number2, number3) && isMathProgression(number2, number3, number4);
+	protected boolean isMultiplier(int multiplier, int number2) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/base/TaskBase.java]: why number2...

File: src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java (hunk 16 above)
Comment (hsp-shant) [reply src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java]: Harout jan, I always test my Tasks, and I retested again, its working, even with debug mode checking.

Comment (hsp-harut) [reply src/main/java/am/hsp/apps/seua/Tasks_151_160/Task_151.java]: Again. You have a bug here. And that is quite bad that you are not taking my comments seriously.
Try to test your code by having a number = 6. The result must be 12, but in your case it is 9. And you are telling that you tested your code? How you tested in that case?

File: src/main/java/am/hsp/apps/seua/tasks_151_160/Task_156.java [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+	@Override
+	public void solve() {
+		System.out.println(getMulipliers());
+	}
+
+	/**
+	 * Print the Multiplication of correct numbers in the range of Tens which are
+	 * Multiple of the number 3 and 5
+	 */
+	private long getMulipliers() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_151_160/Task_156.java]: Wrong method name. Based on your method name - it seems that it should return a sequence of multipliers, and it is even not clear multipliers for what numbers?
Actually your method return a product of multipliers for two digit numbers...

File: src/main/java/am/hsp/apps/seua/tasks_151_160/Task_157.java [hunk 28]
```diff
@@ -0,0 +1,52 @@
...
+
+		System.out.println(sumOfCorrectNumbers());
+
+	}
+
+	/**
+	 * Print the sum of correct numbers in the range of Hundreds which are not Multiple of
+	 * number 5
+	 */
+	private int sumOfCorrectNumbers() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_151_160/Task_157.java]: 1. Wrong method name. It says nothing related the actual logic...
2. Method name MUST BE VERB

File: src/main/java/am/hsp/apps/seua/tasks_151_160/Task_155.java [hunk 29]
```diff
@@ -0,0 +1,49 @@
...
+		
+		System.out.println(sumOfMulipliers());
+
+	}
+
+	/**
+	 * Print the sum of correct numbers in the range of Tens which are Multiple of number 3
+	 * 3
+	 */
+	private int sumOfMulipliers() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_151_160/Task_155.java]: It is not clear your what sum of which multipliers you are calculating? Actually you are calculating sum of multipliers for two digit numbers...

File: src/main/java/am/hsp/apps/seua/tasks_151_160/Task_157.java [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+
+		System.out.println(getSumOfCorrectNumbers());
+
+	}
+
+	/**
+	 * Print the sum of correct numbers in the range of Hundreds which are not Multiple of
+	 * number 5
+	 */
+	private int getSumOfCorrectNumbers() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/tasks_151_160/Task_157.java]: Again wrong method name

Comment (hsp-harut) [reply src/main/java/am/hsp/apps/seua/tasks_151_160/Task_157.java]: You didn't fix my comment again.

**Pull request merged**